jdaviz/configs/cubeviz/plugins/tools.py

```diff
@@ -42,3 +47,150 @@ def on_mouse_event(self, data):
+                                                          label,
+                                                          clear_other_data=False)
+                self.created_data_labels.append(label)
+
+                # Set color of spectrum to not conflict with subset colors
+                for layer in self.spectrum_viewer.state.layers:
+                    if layer.layer.label == label:
+                        layer.color = self.colors[0]
+
+                # Link newly created spectrum to reference data
+                ref_cube = self.spectrum_viewer.state.reference_data
+                new_spec = dc[label]
+                self.link_data(ref_cube, new_spec)
+
+            # Flux viewer is empty or contains only 2D images
+            if not cube_in_viewer:
+                self.remove_highlight_on_spaxel()
+                msg = SnackbarMessage("No cube in viewer, cannot create spectrum from spaxel",
+                                      sender=self, color="warning")
+                self.viewer.session.hub.broadcast(msg)
+                return
+
+    def link_data(self, data1, data2):
+        # If spectrum is also reference data
+        if data1.label == data2.label:
+            return
+
+        world1 = data1.world_component_ids
+        world2 = data2.world_component_ids
+        # Reference data is cube
+        if len(world1) == 3 and len(world2) == 1:
+            links = [LinkSame(world1[2], world2[0])]
+        # Reference data is 1D spectrum
+        elif len(world1) == 1 and len(world2) == 1:
+            links = [LinkSame(world1[0], world2[0])]
+        else:
+            return
+
+        self.viewer.session.data_collection.add_link(links)
+
+    def highlight_spaxel(self, x, y):
+        # Creates an X that shows the spaxel at coordinate (x, y)
+        x_coords, y_coords = self._calc_coords(x, y)
+
+        # Create LinearScale that is the same size as the image viewer
+        scales = {'x': self.viewer.figure.interaction.x_scale,
+                  'y': self.viewer.figure.interaction.y_scale}
+        highlight = SelectedSpaxel(x=x_coords, y=y_coords, scales=scales,
+                                   fill='none', colors=self.colors, stroke_width=2,
+                                   close_path=True)
+        self.viewer.figure.marks = self.viewer.figure.marks + [highlight]
+
+    def update_highlight_spaxel(self, x, y):
+        x_coords, y_coords = self._calc_coords(x, y)
+
+        for spaxel in self.viewer.figure.marks:
+            if isinstance(spaxel, SelectedSpaxel):
+                spaxel.x = x_coords
+                spaxel.y = y_coords
+
+    def remove_highlight_on_spaxel(self):
+        self.viewer.figure.marks = [x for x in self.viewer.figure.marks
+                                    if not isinstance(x, SelectedSpaxel)]
+
+    def _calc_coords(self, x, y):
+        x_coords = [[x - 0.5, x + 0.5], [x - 0.5, x + 0.5]]
+        y_coords = [[y - 0.5, y + 0.5], [y + 0.5, y - 0.5]]
```

Comment:
Pure nitpick. I'll leave it up to you to accept or ignore.
```suggestion
        xmin = x - 0.5
        xmax = x + 0.5
        ymin = y - 0.5
        ymax = y + 0.5
        x_coords = [[xmin, xmax], [xmin, xmax]]
        y_coords = [[ymin, ymax], [ymax, ymin]]
```